src/main/java/racingcar/service/GameService.java

```diff
@@ -1,36 +1,50 @@
 package racingcar.service;
 
-import java.util.HashMap;
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
 import java.util.stream.Collectors;
 
 import org.springframework.stereotype.Service;
 
-import racingcar.dao.CarsDao;
-import racingcar.dao.GamesDao;
-import racingcar.dao.WinnersDao;
 import racingcar.domain.Car;
 import racingcar.domain.Game;
 import racingcar.domain.MoveChance;
 import racingcar.domain.RandomMoveChance;
+import racingcar.repository.GameRepository;
+import racingcar.service.dto.CarDto;
+import racingcar.service.dto.ResultDto;
 
 @Service
 public class GameService {
 
-    private final GamesDao gamesDao;
-    private final CarsDao carsDao;
-    private final WinnersDao winnersDao;
+    private final GameRepository gameRepository;
     private final MoveChance moveChance;
 
-    public GameService(GamesDao gamesDao, CarsDao carsDao, WinnersDao winnersDao) {
-        this.gamesDao = gamesDao;
-        this.carsDao = carsDao;
-        this.winnersDao = winnersDao;
+    public GameService(GameRepository gameRepository) {
+        this.gameRepository = gameRepository;
         this.moveChance = RandomMoveChance.getInstance();
     }
 
-    public Game createGameWith(List<String> names, int trialCount) {
+    public ResultDto playWith(List<String> names, int trialCount) {
+        Game game = createGameWith(names, trialCount);
+        while (game.isNotDone()) {
```

Comment:
```suggestion
        while (game.isRunning()) {
```

지금도 좋지만, 요 네이밍은 어떠신가요?

Reply:
부정적인 네이밍은 추후 부정(!)을 붙였을 때 직관적이지 않을 수 있다고 이해했습니다.
하지만 `isRunning()`은 게임이 별도의 스레드를 가지고 스스로 진행되는 것처럼 보여서, 
`isInProgress()`로 수정해보겠습니다!